src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,119 @@
-import { useState } from 'react';
+import './NewMovie.scss';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const defaultMovie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const NewMovie: React.FC<Props> = ({
+  onAdd,
+}) => {
+  const [count, setCount] = useState(0);
+
+  const [movie, setMovie] = useState(defaultMovie);
+
+  const onInputChange = (name: string, value: string) => {
+    setMovie({
+      ...movie,
+      [name]: value,
+    });
+  };
+
+  const [isImgUrlValid, setIsImgUrlValid] = useState(false);
+  const [isImdbUrlValid, setIsImdbUrlValid] = useState(false);
+
+  const customValidation = (movieToCheck: Movie) => {
+    // eslint-disable-next-line max-len
+    const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w\-_]*)?\??(?:[-+=&;%@.\w_]*)#?(?:[.!/\\\w]*))?)(?:[/,\d_.\w]+)$/gmi;
+
+    if (!movieToCheck.imgUrl.match(pattern)) {
+      setIsImgUrlValid(true);
+    } else {
+      setIsImgUrlValid(false);
+    }
```

Comment:
```suggestion
    setIsImgUrlValid(!!movieToCheck.imgUrl.match(pattern));
```
Let's try this. You can also use the ternary operator if this variant doesn't work